src/cart/crt_context.c

```diff
@@ -958,12 +983,17 @@ crt_context_req_track(struct crt_rpc_priv *rpc_priv)
 void
 crt_context_req_untrack(struct crt_rpc_priv *rpc_priv)
 {
-	struct crt_context	*crt_ctx = rpc_priv->crp_pub.cr_ctx;
-	struct crt_ep_inflight	*epi;
-	int64_t			 credits, inflight;
-	d_list_t		 submit_list;
-	struct crt_rpc_priv	*tmp_rpc;
-	int			 rc;
+	struct crt_context		*crt_ctx = rpc_priv->crp_pub.cr_ctx;
+	struct crt_ep_inflight		*epi;
+	int64_t				 credits, inflight;
+	d_list_t			 submit_list;
+	struct crt_rpc_priv		*tmp_rpc;
+	int				 rc;
+	static struct d_tm_node_t	*request_counter;
+	static struct d_tm_node_t	*abort_counter;
+	static struct d_tm_node_t	*wait_queue_gauge;
+	static struct d_tm_node_t	*reply_counter;
+	static struct d_tm_node_t	*inflight_queue_gauge;
```

Comment:
Why not putting all of this in the context instead?

Reply:
My intention was that these were global across all contexts.  I see your point that having these local to a context would be helpful.  There's one consideration with that, that I need to work through first.  Is there a fixed number of these contexts for the life of the IO server?  If the number is fixed, then it's straight forward to add these values to the context.  If the number grows over time as contexts come and go, then there's a memory management issue to work through to reclaim the shared memory that was allocated to an old unused context to avoid running out of shared memory as new contexts (with new data sets) are added.  Can you advise about the nature of the context?